src/app/layout.tsx

```diff
@@ -19,8 +22,12 @@ export default function RootLayout({
       className={`${font_body.variable} ${font_heading.variable}`}
     >
       <body>
-        <Header />
-        {children}
+      <Providers>
+        <Suspense fallback={<Spinner/>}>
+          <Header />
+          {children}
+        </Suspense>
```

Comment:
hmm, I think we shouldn't wrap the whole layout inside `Suspense` Bishwas.

`Suspense` is usually to show fallback for the section that takes long time to fetch data or load. Thus, it's mostly beneficial and used for each granular section. Right now, the whole page will have loading spinner which is not something we want.

The docs is really helpful. https://react.dev/reference/react/Suspense Let me know if you have any queries.

Reply:
`Suspense` here is for Streaming Server Rendering when content of a route segment loads. It is not being used on async operation or to  fetch data here. That part of Suspense is completely different.

Reply:
Removed `Suspense` here as loading in route segment change is handled by `loading` file in nextjs
https://nextjs.org/docs/app/building-your-application/routing/loading-ui-and-streaming